Core/MIPS/IR/IRInterpreter.cpp

```diff
@@ -304,13 +304,17 @@ u32 IRInterpret(MIPSState *mips, const IRInst *inst, int count) {
 		}
 
 		case IROp::Vec4Shuffle:
-		{
 			// Can't use the SSE shuffle here because it takes an immediate. pshufb with a table would work though,
 			// or a big switch - there are only 256 shuffles possible (4^4)
 			for (int i = 0; i < 4; i++)
 				mips->f[inst->dest + i] = mips->f[inst->src1 + ((inst->src2 >> (i * 2)) & 3)];
 			break;
-		}
+
+		case IROp::Vec4Blend:
+			// Could use _mm_blendv_ps (SSE4+BMI), vbslq_f32 (ARM), __riscv_vmerge_vvm (RISC-V)
```

Comment:
_mm_blendv_ps is a bit annoying, since turning the 4 bits into a 128-bit mask register (it looks at the top bit of every dword, instead of packed bits, unfortunately) would require lookup into a table of such masks. Which, to be fair, might be alright..

Reply:
Yeah.  For jit, it'd be fine.  It might be worth doing here, not sure.  I think using PDEP + VBLENDV would be most optimal.

For BSL, it's basically the same thing (it selects each bit), but RISC-V's vmerge is actually low bits.

-[Unknown]